core/src/main/java/io/kestra/core/plugins/PluginResolver.java

```diff
@@ -35,7 +40,32 @@ private static boolean isClassFile(final Path path) {
         return path.toString().toLowerCase().endsWith(".class");
     }
 
-    public List<ExternalPlugin> resolves() {
+    public void watch(Consumer<ExternalPlugin> onNewPlugin) throws Exception {
+        this.initialPlugins().forEach(onNewPlugin);
+        watchService = FileSystems.getDefault().newWatchService();
+        pluginPath.register(watchService, StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY);
```

Comment:
We should handle also the delete no ? to remove the plugin that will be removed ?

Reply:
I went for it at first but it would need to map every fileName to its ExternalPlugin because the ENTRY_DELETE event occurs once the file is removed so there is no way afterwards to know which plugin to remove.
I talked about it with others and they told me to keep things very simple for now so I went ahead and removed it :thinking:

Reply:
If you think it's worth it I can spend more time making it work... Or we can create another issue for it :thinking:

Reply:
we need to have some consistency and without delete the users would need to restart its clusters (what we don't want)